tests/test_session_filtering.py

```
@@ -111,32 +115,305 @@ def sample_traces(self):
-    def test_filter_by_has_media(self, client, sample_traces):
+    def test_filter_by_has_media(self, client, sample_traces, mock_db_session):
         """Test filtering traces by media presence."""
-        pytest.skip("Skipping unit test - covered by integration tests")
+        from src.api.deps import get_database
+        from unittest.mock import Mock
+        
+        # Mock the database dependency
+        client.app.dependency_overrides[get_database] = lambda: mock_db_session
+        
+        # Configure mock to return only traces with media
+        expected_traces = [trace for trace in sample_traces if trace.has_media == True]
```

Ruff (0.12.2)

Avoid equality comparisons to `True`; use `trace.has_media:` for truth checks

Replace with `trace.has_media`

(E712)